src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -232,14 +237,72 @@ protected void onStartup() {
         super.onStartup();
 
         initializeReference();
+
         initializeReads();
+
         initializeFeatures();
+
         initializeIntervals(); // Must be initialized last, since intervals currently require a sequence dictionary from another data source
 
+        validateSequenceDictionaries();
+
         checkToolRequirements();
     }
 
     /**
+     * Validates all sequence dictionaries by checking them against each other.
+     *
+     * Currently, read dict is checked against
+     * ref dict, and both are checked against any variant header dicts. No interval dicts are checked at this time.
+     *
+     */
+    private void validateSequenceDictionaries() {
+        final SAMSequenceDictionary refDict = hasReference() ? reference.getSequenceDictionary() : null;
+        final SAMSequenceDictionary readDict = hasReads() ? reads.getSequenceDictionary() : null;
+        Hashtable<String,SAMSequenceDictionary> variantDicts = new Hashtable<>();
```

Comment:
Actually you should not be using a hash here at all -- just create an `ArrayList` of the variant dictionaries, and get the ones you want by index rather than by name. This is faster than a hash lookup.